activesupport/lib/active_support/message_pack.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+begin
+  require "msgpack"
+rescue LoadError => error
+  $stderr.puts "You don't have msgpack installed in your application. " \
+    "Please add it to your Gemfile and run bundle install."
+  raise error
+end
+
+require_relative "message_pack/extensions"
+require_relative "message_pack/serializer"
+
+module ActiveSupport
+  module MessagePack
+    extend Serializer
+    self.message_pack_factory = Extensions.configure_factory(::MessagePack::Factory.new)
```

Comment:
protip: you wanna freeze the factory once you're done registring types. It allows to share some internal state with packers/unpackers instead of copying it.

Reply:
Nice!  Thanks! :smile: